designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/StatisticsUrlsDomainService.scala

```diff
@@ -0,0 +1,89 @@
+package pl.touk.nussknacker.ui.statistics
+
+import cats.implicits.toTraverseOps
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.ui.statistics.StatisticsUrlsDomainService.{
+  emptyString,
+  nuStatsUrl,
+  queryParamsSeparator,
+  toURL,
+  urlBytesSizeLimit
+}
+
+import java.net.{URI, URL, URLEncoder}
+import java.nio.charset.StandardCharsets
+import scala.collection.mutable.ArrayBuffer
+import scala.util.{Failure, Success, Try}
+
+class StatisticsUrlsDomainService(private val fingerprint: Fingerprint, private val statistics: Map[String, String])
+    extends LazyLogging {
+
+  private lazy val queryParamsForEveryURL = List(
+    encodeQueryParam(NuFingerprint.name -> fingerprint.value)
+  )
+
+  def prepareUrls(): Either[StatisticError, List[URL]] = {
```

Comment:
WDYT about doing the conversion (from the Statistics class to the List of URLs) in the extension class (it may be defined in the compaction object)? 

Currently, we create URLs from the statistics, but maybe in the future, we will do it differently. So, the URL creation logic could be separated from the data class.

Reply:
Done